### DISCUSSION TITLE
`onError` handler in custom `baseQuery`

### DISCUSSION THREAD

**Discussion by abhieshekumar** [reduxjs/redux-toolkit#3099] [Q&A] [answered]
Hi, the current behaviour is that the `onError` callback is executed each time after a retry fails.

```js
query:() => ({
  url: API_PATH,
  method: "GET",
  // execute once all retries have failed too. Right now it is getting executed for each retry
  onError: ()=>{
    showNotification(ERROR);
  }
})
extraOptions: { maxRetries: NUMBER },
```

Is there a recommended approach to only execute the callback once all the retries have failed?

**Comment by phryneas** [comment, marked as answer]
There is no such thing as an `onError` callback in RTK Query. I'm honestly a bit confused right now. Are you using a custom `baseQuery`?

**Reply by abhieshekumar** [reply]
my bad, yes it's a custom baseQuery.

**Reply by phryneas** [reply]
My advice would be not to have it be part of your `baseQuery`, because `retry` wraps around that - it cannot look inside. Use the `onQueryStarted` lifecycle event instead, you can handle errors there.

**Reply by abhieshekumar** [reply]
thanks @phryneas
